test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -739,4 +739,29 @@ TEST_VM_F(NMTVMATreeTest, SummaryAccountingWhenUseFlagInplace) {
   diff = tree.uncommit_mapping(0, 50, rd2);
   EXPECT_EQ(0, diff.tag[NMTUtil::tag_to_index(mtTest)].reserve);
   EXPECT_EQ(-50, diff.tag[NMTUtil::tag_to_index(mtTest)].commit);
+}
+
+TEST_VM_F(NMTVMATreeTest, UncommmitReleasedRegion) {
+  {
```

Comment:
I would also add this case (which is the one I used originally to find this issue):

```
    Tree tree;
    // 0..............................
    // 0.......40----60........
    VMATree::SummaryDiff diff = tree.uncommit_mapping(40, 20, rd2);
    EXPECT_FALSE(diff.is_valid())z
```

Reply:
Thanks for reminding. Handling this case is also added.